ocean_lib/models/v4/test/test_factory_router.py

```diff
@@ -0,0 +1,543 @@
+#
+# Copyright 2021 Ocean Protocol Foundation
+# SPDX-License-Identifier: Apache-2.0
+#
+from web3.main import Web3
+from tests.resources.helper_functions import (
+    get_publisher_wallet,
+    get_consumer_wallet,
+    get_another_consumer_wallet,
+)
+
+import pytest
+
+from web3 import exceptions
+from ocean_lib.models.v4.erc721_factory import ERC721FactoryContract
+from ocean_lib.models.v4.factory_router import FactoryRouter
+from ocean_lib.utils.addresses_utils import (
+    get_nft_factory_address,
+    get_factory_router_address,
+    get_ocean_address,
+    get_pool_template_address,
+    get_staking_address,
+    get_erc20_template_address,
+    get_fixed_price_address,
+)
+from ocean_lib.models.v4.erc20_token import ERC20Token
+from tests.resources.helper_functions import get_factory_deployer_wallet
+from ocean_lib.web3_internal.constants import ZERO_ADDRESS
+
+_NETWORK = "ganache"
+
+
+def create_new_token(web3, config):
+    erc20_token = ERC20Token.deploy(web3=web3, deployer_wallet=get_publisher_wallet())
+    return erc20_token
+
+
+def test_properties(web3, config):
+    """Tests the events' properties."""
+
+    factory_router = FactoryRouter(
+        web3,
+        get_factory_router_address(config),
+    )
+    assert factory_router.event_NewPool.abi["name"] == FactoryRouter.EVENT_NEW_POOL
+
+
+def test_ocean_tokens_mapping(web3, config):
+    """Tests that Ocean token has been added to the mapping"""
+    factory_router = FactoryRouter(
+        web3,
+        get_factory_router_address(config),
+    )
+    ocean_tokens = factory_router.ocean_tokens(get_ocean_address(config))
+    assert ocean_tokens is True
+
+
+def test_add_token(web3, config):
+    """Tests adding a new token address to the mapping if Router Owner"""
+    new_token_address = ERC20Token.deploy(
+        web3=web3, deployer_wallet=get_factory_deployer_wallet(network=_NETWORK)
+    )
+    factory_router = FactoryRouter(
```

Comment:
let's move this in the conftest file, as a fixture and use it as an argument. No point in doing the same initializing multiple times. Or set it at the beginning of the file, as a variable or a fixture.